### DISCUSSION TITLE
[closed][manuscript][x][outreach] How should we handle versioning of ontologies?

### DISCUSSION THREAD

**Discussion by sveinugu** [fairtracks/fairtracks_standard#112] [Team Posts]
Should we always link to the current version of an ontology, meaning that what is allowed by our standard will fluctuate according to the external ontologies being modified? Or should we link to a particular version of external ontologies, and update our standard regularly (going through all referenced ontologies)? Which also relates to how should we version our JSON schema itself?

I did discuss this point with Matus Kalas in the EDAM project, who had very clear overview of the issues here. However, I do not remember the details (I don't think I wrote them down)..

**Comment by sveinugu** [comment]
Sent an email to Matus and asked about a recap of the arguments regarding ontology versioning

**Comment by sveinugu** [comment]
Will meet Matus F2F next week due to Elixir Norway all-hands, and we will discuss the issue, and possibly our suggestions to https://github.com/fairtracks/fairtracks_standard/issues/5 and https://github.com/fairtracks/fairtracks_standard/issues/11, as well.

**Comment by jmfernandez** [comment]
It's a tricky question. On one hand, being tied to an specific version assures all the previously validated terms are still valid. On the other hand, if we are on the bleeding edge of the ontology, and its maintainers do not keep the IRIs of the deprecated or removed terms in some way, previously validated terms may become invalid, leading to previously valid FAIRTracks to become invalid.

On the technical side, the validator does not have the needed logic to check whether an ontology should be re-fetched (maybe the library I'm using Owlready2, but not sure).

**Comment by sveinugu** [comment]
@jmfernandez The lack of logic in the validator would be if we do not version control the ontologies, right? If we do version them, you could just re-fetch the updated ontologies whenever you load a new version of the standard.

**Comment by sveinugu** [comment]
Added issue: https://github.com/fairtracks/fairtracks_standard/issues/22. Closing this

**Comment by sveinugu** [comment]
Added link in https://github.com/orgs/fairtracks/teams/fairtracks-developers/discussions/41.
